src/OpenSpaceToolkit/Astrodynamics/EventCondition.cpp

```diff
@@ -0,0 +1,136 @@
+{
+    anEventCondition.print(anOutputStream);
+
+    return anOutputStream;
+}
+
+String EventCondition::getName() const
+{
+    return name_;
+}
+
+EventCondition::Criteria EventCondition::getCriteria() const
+{
+    return criteria_;
+}
+
+void EventCondition::print(std::ostream& anOutputStream, bool displayDecorator) const
+{
+    displayDecorator ? ostk::core::utils::Print::Header(anOutputStream, "Event Condition") : void();
+
+    ostk::core::utils::Print::Line(anOutputStream) << "Name:" << getName();
+    ostk::core::utils::Print::Line(anOutputStream) << "Criteria:" << StringFromCriteria(getCriteria());
+
+    displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void();
+}
+
+bool EventCondition::isSatisfied(const Real& currentValue, const Real& previousValue) const
+{
+    return comparator_(currentValue, previousValue);
+}
+
+String EventCondition::StringFromCriteria(const Criteria& aCriteria)
+{
+    switch (aCriteria)
+    {
+        case Criteria::PositiveCrossing:
+            return "Positive Crossing";
+
+        case Criteria::NegativeCrossing:
+            return "Negative Crossing";
+
+        case Criteria::AnyCrossing:
+            return "Any Crossing";
+
+        case Criteria::PositiveOnly:
+            return "Positive Only";
+
+        case Criteria::NegativeOnly:
+            return "Negative Only";
+
+        case Criteria::Undefined:
+            return "Undefined";
+
+        default:
+            throw ostk::core::error::runtime::Wrong("Criteria");
+    }
+
+    return String::Empty();
+}
+
+std::function<bool(const Real&, const Real&)> EventCondition::getComparator(const EventCondition::Criteria& aCriteria)
+{
+    switch (aCriteria)
+    {
+        case EventCondition::Criteria::PositiveOnly:
+            return [](const Real& currentValue, const Real& previousValue) -> bool
+            {
+                (void)previousValue;
+                return currentValue > 0.0;
+            };
+
+        case EventCondition::Criteria::NegativeOnly:
+            return [](const Real& currentValue, const Real& previousValue) -> bool
+            {
+                (void)previousValue;
+                return currentValue < 0.0;
+            };
+
+        case EventCondition::Criteria::PositiveCrossing:
+            return [](const Real& currentValue, const Real& previousValue) -> bool
+            {
+                return previousValue < 0.0 && currentValue > 0.0;
+            };
+
+        case EventCondition::Criteria::NegativeCrossing:
+            return [](const Real& currentValue, const Real& previousValue) -> bool
+            {
+                return previousValue > 0.0 && currentValue < 0.0;
+            };
+
+        case EventCondition::Criteria::AnyCrossing:
+            return [](const Real& currentValue, const Real& previousValue) -> bool
+            {
+                return ((currentValue > 0.0) == (previousValue < 0.0));
```

Comment:
```suggestion
                return (currentValue > 0.0) == (previousValue < 0.0);
```

nit. idk what the formatter thinks about this though

Reply:
Let me try!

Reply:
Works!